nca/Parsers/IstioVirtualServiceYamlParser.py

```diff
@@ -0,0 +1,600 @@
+        """
+        result = []
+        global_has_mesh = 'mesh' in vs.gateway_names
+        has_gateways = len(vs.gateway_names) > 1 if global_has_mesh else bool(vs.gateway_names)
+        local_peers = self.peer_container.get_all_peers_group()
+        egress_gtw_pods = self.gtw_parser.get_egress_gtw_pods()
+        mesh_to_egress_policy = self.init_mesh_to_egress_policy(vs, egress_gtw_pods)
+        mesh_to_mesh_warning_printed = False  # to avoid multiple printing of this warning
+
+        for route_cnt, route in enumerate(routes, start=1):
+            if route.gateway_names:  # override global gateways
+                has_mesh = 'mesh' in route.gateway_names
+                has_gateways |= len(route.gateway_names) > 1
+                gtw_to_hosts = self.pick_vs_gateways_by_hosts(vs, route.gateway_names)
+            else:  # use global gateways
+                has_mesh = global_has_mesh
+                gtw_to_hosts = global_vs_gtw_to_hosts
+            if not has_mesh and not gtw_to_hosts:  # when no gateways are given, the default is mesh
+                has_mesh = True
+            if route.is_internal_dest and has_mesh and not mesh_to_mesh_warning_printed:
+                self.warning(f'The internal (mesh-to-mesh) traffic redirection mentioned in the '
+                             f'VirtualService {vs.full_name()} is not currently supported and will be ignored', vs)
+                mesh_to_mesh_warning_printed = True
+            result.extend(self.create_gtw_to_mesh_policies(vs, route, route_cnt, gtw_to_hosts, used_gateways))
+            # Modeling connections from mesh to (egress) gateway nodes (which should be identified) (Egress flow).
+            # Not modelling other connections from mesh to internal nodes here.
+            # Modeling deny all connections from mesh to external service nodes (DNS nodes) (Egress flow).
+            this_route_conn_cube = self.init_route_conn_cube(route)
+            for dest in route.destinations:
+                if not route.is_internal_dest:
+                    continue  # external dest cannot be an egress gateway pod
+                if has_mesh and dest.pods.issubset(egress_gtw_pods):
+                    # add a rule to mesh_to_egress_policy
+                    this_dest_conn_cube = this_route_conn_cube.copy()
+                    this_dest_conn_cube["dst_ports"] = dest.ports
+                    conns = ConnectionSet()
+                    conns.add_connections(self.protocol_name, ConnectivityProperties.make_conn_props(this_dest_conn_cube))
+                    this_dest_conn_cube.update({"src_peers": local_peers, "dst_peers": dest.pods,
+                                                "protocols": self.protocols})
+                    opt_props = ConnectivityProperties.make_conn_props(this_dest_conn_cube)
+                    mesh_to_egress_policy.add_ingress_rule(GatewayPolicyRule(local_peers, conns, opt_props))
```

Comment:
for example, this can be in a separate function

Reply:
As a part of code reorganization this is moved to another function